src/modules/configuration/src/lib/Configuration.c

```diff
@@ -0,0 +1,545 @@
+{
+    MMI_HANDLE handle = (MMI_HANDLE)g_configurationModuleName;
+    g_maxPayloadSizeBytes = maxPayloadSizeBytes;
+    ++g_referenceCount;
+    OsConfigLogInfo(ConfigurationGetLog(), "MmiOpen(%s, %d) returning %p", clientName, maxPayloadSizeBytes, handle);
+    return handle;
+}
+
+static bool IsValidSession(MMI_HANDLE clientSession)
+{
+    return ((NULL == clientSession) || (0 != strcmp(g_configurationModuleName, (char*)clientSession)) || (g_referenceCount <= 0)) ? false : true;
+}
+
+void ConfigurationMmiClose(MMI_HANDLE clientSession)
+{
+    if (IsValidSession(clientSession))
+    {
+        --g_referenceCount;
+        OsConfigLogInfo(ConfigurationGetLog(), "MmiClose(%p)", clientSession);
+    }
+    else
+    {
+        OsConfigLogError(ConfigurationGetLog(), "MmiClose() called outside of a valid session");
+    }
+}
+
+int ConfigurationMmiGetInfo(const char* clientName, MMI_JSON_STRING* payload, int* payloadSizeBytes)
+{
+    int status = EINVAL;
+
+    if ((NULL == payload) || (NULL == payloadSizeBytes))
+    {
+        return status;
+    }
+    
+    *payload = NULL;
+    *payloadSizeBytes = (int)strlen(g_configurationModuleInfo);
+    
+    *payload = (MMI_JSON_STRING)malloc(*payloadSizeBytes);
+    if (*payload)
+    {
+        memcpy(*payload, g_configurationModuleInfo, *payloadSizeBytes);
+        status = MMI_OK;
+    }
+    else
+    {
+        OsConfigLogError(ConfigurationGetLog(), "MmiGetInfo: failed to allocate %d bytes", *payloadSizeBytes);
+        *payloadSizeBytes = 0;
+        status = ENOMEM;
+    }
+    
+    if (IsFullLoggingEnabled())
+    {
+        OsConfigLogInfo(ConfigurationGetLog(), "MmiGetInfo(%s, %.*s, %d) returning %d", clientName, *payloadSizeBytes, *payload, *payloadSizeBytes, status);
+    }
+
+    return status;
+}
+
+int ConfigurationMmiGet(MMI_HANDLE clientSession, const char* componentName, const char* objectName, MMI_JSON_STRING* payload, int* payloadSizeBytes)
+{
+    int status = MMI_OK;
+    char buffer[10] = {0};
+    char* configuration = NULL;
+
+    if ((NULL == componentName) || (NULL == objectName) || (NULL == payload) || (NULL == payloadSizeBytes))
+    {
+        OsConfigLogError(ConfigurationGetLog(), "MmiGet(%s, %s, %p, %p) called with invalid arguments", componentName, objectName, payload, payloadSizeBytes);
+        status = EINVAL;
+        return status;
+    }
+
+    *payload = NULL;
+    *payloadSizeBytes = 0;
+
+    if (!IsValidSession(clientSession))
+    {
+        OsConfigLogError(ConfigurationGetLog(), "MmiGet(%s, %s) called outside of a valid session", componentName, objectName);
+        status = EINVAL;
+    }
+    
+    if ((MMI_OK == status) && (strcmp(componentName, g_configurationComponentName)))
+    {
+        OsConfigLogError(ConfigurationGetLog(), "MmiGet called for an unsupported component name (%s)", componentName);
+        status = EINVAL;
+    }
+    
+    if ((MMI_OK == status) && (NULL == (configuration = LoadConfigurationFromFile(g_configurationFile))))

```

Comment:
Prefer to keep these two separate in this case. If there would be a larger number of checks, it may make sense to group using that pattern. In this case choice is not to.